### PR TITLE
Warn on new use of param shortcode

### PR DESCRIPTION
**Pull request by sftim**
[AIUI] we prefer not to use the `param` shortcode (we switched to a new approach in PR https://github.com/kubernetes/website/pull/40692)

Change that shortcode to warn if it's used other than for `version` (which we still support, although I recommend using the `skew` shortcode in its place).
~As part of this:~
- ~switch Bengali to the new approach~
- ~fix a broken link in a blog article~
- ~change a couple of pages to use the `skew` shortcode~

/area web-development

### CODE REVIEW

File: content/bn/docs/tasks/tools/install-kubectl-windows.md
Comment (sftim) [inline content/bn/docs/tasks/tools/install-kubectl-windows.md:27]: This is a machine-aided translation; I don't speak Bengali

File: layouts/shortcodes/param.html
Comment (dipesh-rawat) [inline layouts/shortcodes/param.html:1]: I'm surprised that removing the shortcode HTML file hasn't caused build errors, considering there are still instances in the repo where the `<param>` shortcode is used ([example 1](https://github.com/kubernetes/website/blob/ea4444a8498bab841af7ea1dd56ffb35a9247d32/content/en/docs/concepts/workloads/pods/_index.md?plain=1#L224), [example 2](https://github.com/kubernetes/website/blob/d8e55f2ec1d66d115860d80c4011de5965c90881/content/en/releases/download.md?plain=1#L56)). I assume now the site is using the shortcode source retrieved from Docsy or its dependencies during the build process.

**Review by dipesh-rawat — COMMENTED** [review]
Overall, the change looks good! 
_Just added an observation on the removal of shortcode._

LGTM

**Comment by sftim** [comment]
/retitle Drop param shortcodes, warn on new use

**Comment by sftim** [comment]
Do we want to merge this with lots of pages showing warnings, or split out the Bengali fixes, or something else?

**Comment by sftim** [comment]
/retitle Fix legacy param shortcodes; warn on new use

File: content/bn/docs/tasks/tools/install-kubectl-windows.md
Comment (asem-hamid) [inline content/bn/docs/tasks/tools/install-kubectl-windows.md:27]:
```suggestion
1. সর্বশেষ {{< skew currentVersion >}} প্যাচ রিলিজ ডাউনলোড করুন: [kubectl {{% skew currentPatchVersion %}}](https://dl.k8s.io/release/v{{% skew currentPatchVersion %}}/bin/windows/amd64/kubectl.exe)।
```

**Review by asem-hamid — CHANGES_REQUESTED** [review]

File: content/bn/docs/tasks/tools/install-kubectl-windows.md
Comment (asem-hamid) [inline content/bn/docs/tasks/tools/install-kubectl-windows.md:27]:
```suggestion
1. সর্বশেষ {{< skew currentVersion >}} প্যাচ রিলিজ ডাউনলোড করুন: [kubectl {{% skew currentPatchVersion %}}](https://dl.k8s.io/release/v{{% skew currentPatchVersion %}}/bin/windows/amd64/kubectl.exe)।
```

**Review by asem-hamid — CHANGES_REQUESTED** [review]

**Comment by dipesh-rawat** [comment]
> Do we want to merge this with lots of pages showing warnings, or split out the Bengali fixes, or something else?

Focusing this PR specifically on Bengali fixes makes a lot of sense. For the other changes related to shortcode and its usage in English doc pages and blogs, let's include those in a separate PR. This will help in review and also adhere more closely to the practice of making changes to a single language per PR.

**Comment by ydFu** [comment]
> Do we want to merge this with lots of pages showing warnings, or split out the Bengali fixes, or something else?

It is recommended to ask reviewers from different languages ​​to review the changed localization content before merging it.

**Comment by ricardoamaro** [comment]
I have reviewed this pull request. 
It is well-structured and moves towards a more maintainable and standardized way of handling version references and other parameters within the site's content.
I agree that reviewers from different languages should look at the changed localization content before merging it.

**Comment by sftim** [comment]
/hold

Let's rebase this on _main_ once language-specific fixes have merged in.

**Comment by asem-hamid** [comment]
bn fixes has been merged in this [PR](https://github.com/kubernetes/website/pull/46346)

**Comment by sftim** [comment]
/retitle Warn on new use of param shortcode

File: layouts/shortcodes/param.html
Comment (sftim) [inline layouts/shortcodes/param.html:4]: This is Hugo's default `param` shortcode.

File: layouts/shortcodes/param.html
Comment (sftim) [inline layouts/shortcodes/param.html:13]: This is our warning mechanism.

**Comment by sftim** [comment]
Updated. This PR won't add warnings, but future pages that use `param` might make a new warning show up.

**Comment by sftim** [comment]
/hold cancel

The fixes to Bengali have merged.

**Review by dipesh-rawat — COMMENTED** [review]
The changes look good!

/lgtm

**Comment by ydFu** [comment]
/lgtm
This change can solve specific problems and will not involve files with cross-language permissions.

**Pull request merged**
